aplazame/controllers/front/Api/confirm.php

```diff
@@ -91,18 +97,22 @@ public function confirm($payload)
             return self::ko('Fraud detected');
         }
 
+        $newMid = null;
+
         switch ($payload['status']) {
             case 'pending':
                 switch ($payload['status_reason']) {
                     case 'challenge_required':
                         if (!$this->module->pending($cart)) {
                             return self::ko("'pending' function failed");
                         }
+                        $newMid = $id_cart ? $this->buildMid($cartId) : null;
```

Comment:
haría `return self::ok($this->buildMid($cartId))` para simplificar

Reply:
Hecho, aunque le he pasado también a la función el `$queryArguments` para comprobar dentro de la misma si está definido `cart_id` para mantener la retrocompatibilidad y no cambiar el `mid` de un pedido sin `cart_id` en su `notification_url`.